internal/controlplane/handlers_githubwebhooks.go

```diff
@@ -113,6 +113,69 @@ func entityFromWebhookEventTypeKey(m *message.Message) pb.Entity {
 	return pb.Entity_ENTITY_UNSPECIFIED
 }
 
+// HandleGitHubAppWebhook handles incoming GitHub App webhooks
+func (s *Server) HandleGitHubAppWebhook() http.HandlerFunc {
+	return func(w http.ResponseWriter, r *http.Request) {
+		wes := &metrics.WebhookEventState{
+			Typ:      "unknown",
+			Accepted: false,
+			Error:    true,
+		}
+		defer func() {
+			s.mt.AddWebhookEventTypeCount(r.Context(), wes)
+		}()
+
+		rawWBPayload, err := validatePayloadSignature(r, &s.cfg.WebhookConfig)
+		if err != nil {
+			log.Printf("Error validating webhook payload: %v", err)
+			w.WriteHeader(http.StatusBadRequest)
+			return
+		}
+
+		wes.Typ = github.WebHookType(r)
+		if wes.Typ == "ping" {
+			log.Printf("ping received")
+			wes.Error = false
+			return
+		}
+
+		m := message.NewMessage(uuid.New().String(), nil)
+		m.Metadata.Set(events.ProviderDeliveryIdKey, github.DeliveryID(r))
+		m.Metadata.Set(events.ProviderSourceKey, "https://api.github.com/") // TODO: handle other sources
+		m.Metadata.Set(events.GithubWebhookEventTypeKey, wes.Typ)
+
+		ctx := r.Context()
+		l := zerolog.Ctx(ctx).With().
+			Str("webhook-event-type", m.Metadata[events.GithubWebhookEventTypeKey]).
+			Str("providertype", m.Metadata[events.ProviderTypeKey]).
+			Str("upstream-delivery-id", m.Metadata[events.ProviderDeliveryIdKey]).
+			Logger()
```

Comment:
Do you want extract these values on 129, and be able to use `l` on 137 and beyond?

Also, a slight preference for `l := zerolog.Ctx(r.Context())...`, rather than adding a temp value whose lifetime I need to reason about.